File: 0540-single-element-in-a-sorted-array/0540-single-element-in-a-sorted-array.java
class Solution {
    public int singleNonDuplicate(int[] nums) {
        if (nums.length == 1) return nums[0];
        
        for (int i = 1; i < nums.length; ){
            if(nums[i] == nums[i-1]){
                i += 2;
            } else {
                return nums[i-1];
            }
        }
        
        return nums[nums.length - 1]; 
    }
}

File: 0540-single-element-in-a-sorted-array/0540-single-element-in-a-sorted-array.cpp
class Solution {
public:
    int singleNonDuplicate(vector<int>& nums) {
        if(nums.size()==1) return nums[0];
        
        for(int i=1;i<nums.size();){
            if(nums[i]==nums[i-1]){
                i+=2;
            } else {
                return nums[i-1];
            }
        }
        
        return nums[nums.size()-1];
    }
};